addons/pos_self_order/controllers/entry.py

```diff
@@ -30,39 +26,38 @@ class PosQRMenuController(http.Controller):
             ],
         auth="public", website=True, sitemap=True,
     )
-    def pos_self_order_start(self, config_id=None, access_token=None, table_identifier=None):
+    def pos_self_order_start(self, config_id=None, access_token=None):
+        return request.render(
+            'pos_self_order.index',
+            self._prepare_self_order_data(config_id, access_token),
+        )
+
+    def _prepare_self_order_data(self, config_id=None, access_token=None):
         if config_id.isnumeric():
             pos_config_sudo = request.env["pos.config"].sudo().search([
                 ("id", "=", config_id),
                 ('access_token', '=', access_token)], limit=1)
 
         self_order_mode = 'qr_code'
-        table_infos = False
         pos_config_access_token = False
 
-        if pos_config_sudo and pos_config_sudo.has_active_session and pos_config_sudo.self_order_table_mode:
+        if pos_config_sudo and pos_config_sudo.has_active_session and pos_config_sudo.self_order_ordering_mode:
             self_order_mode = pos_config_sudo.self_order_pay_after
             pos_config_access_token = pos_config_sudo.access_token
-            table_sudo = get_table_sudo(identifier=table_identifier)
-            table_infos = table_sudo._get_self_order_data() if table_sudo else False
         else:
             pos_config_sudo = get_any_pos_config_sudo()
 
-        return request.render(
-            'pos_self_order.index',
-            {
+        return {
                 'session_info': {
                     **request.env["ir.http"].get_frontend_session_info(),
                     'currencies': request.env["ir.http"].get_currencies(),
                     'pos_self_order_data': {
                         'self_order_mode': self_order_mode,
-                        'table': table_infos,
                         'access_token': pos_config_access_token,
                         **pos_config_sudo._get_self_order_data(),
                     },
                 }
             }
-        )
 
     @http.route(
```

Comment:
rename the endpoint to `/self-order/get-image` to accomodate kiosk later. (Not so important)